sklearn/preprocessing/label.py

```diff
@@ -472,9 +473,23 @@ def label_binarize(y, classes, neg_label=0, pos_label=1, sparse_output=False):
 
     if y_type == "binary":
         if len(classes) == 1:
-            Y = np.zeros((len(y), 1), dtype=np.int)
-            Y += neg_label
-            return Y
+            if sparse_output:
+                if (isinstance(classes[0], (numbers.Integral, np.integer))
+                        and classes[0] != 0):
+                    classes = np.r_[0, classes]
+                    n_classes += 1
+                else:
+                    classes = np.append(classes, 1)
+                    n_classes += 1
+            else:
+                Y = np.zeros((len(y), 1), dtype=np.int)
+                if (isinstance(classes[0], (numbers.Integral, np.integer))
+                        and classes[0] != 0):
+                    Y += pos_label
+                    return Y
+                else:
+                    Y += neg_label
+                    return Y
```

Comment:
Actually even I think we should default to the positive label in this case but then I'll have to modify the original tests. Should I do that?